### DISCUSSION TITLE
Can FromQuery get the name attr?

### DISCUSSION THREAD

**Discussion by 56907dzq** [Neoteroi/BlackSheep#266] [Ideas]
I notice that we can use FromQuery like this:
`search: FromQuery[str] = FromQuery("")`
FromQuery have two attribute, value and name.
we can get search'value from `search.value`
but `search.name` returned None.
if we can get` search.name =="search"`, I think it's cool!
[FromQuery](https://github.com/Neoteroi/BlackSheep/blob/9fe7d66baf13456e3725f66f41cbc961285a8e2d/blacksheep/server/bindings.py#L147)

**Comment by RobertoPrevato** [comment]
This is a very good question. I need to improve the type annotations of the `BoundValue[T]` class.

The nullable `name` declared in the class is meant to be a class attribute, not a instance attribute. This is necessary when the parameter name cannot be described by a Python parameter name (e.g. header name "X-Something").

For example:

```python
class FromFunnyQuery(FromQuery[str]):
    name = "x-funny-value"


@app.route("/")
async def home(foo: FromFunnyQuery = FromFunnyQuery("")):
    print(foo.name)
    return f"{foo.value=}"
```

This will read requests such as: `http://localhost/?x-funny-value=hello%20world` (here foo has value "hello world").

Other example with cookie and header binders:

```python
from typing import Optional

from blacksheep import Application, Response, text, FromHeader, FromCookie

app = Application()
get = app.router.get


class FromAcceptHeader(FromHeader[str]):
    name = "Accept"


class FromFooCookie(FromCookie[Optional[str]]):
    name = "foo"


@get("/")
def home(accept: FromAcceptHeader, foo: FromFooCookie) -> Response:
    return text(
        f"""
        Accept: {accept.value}
        Foo: {foo.value}
        """
    )
```
